### DISCUSSION TITLE
How to update Teslamate when running on a Google Cloud VM?

### DISCUSSION THREAD

**Discussion by e-tplus** [teslamate-org/teslamate#3768] [General]
Hello,
1.28.3 (and previous) runs fine but when I try to update to 1.28.4 it does not start. Seems that's because of an old Docker version (19.03.12). 

Teslamate is running on a Google e2-micro machine, OS is Debian 9 (Stretch). It's running for some time, now, and backup-file is huge. 

What's the most feasible way to upgrade?

Thanks for some help...

**Comment by cwanja** [comment]
You will need to update Docker.

**Comment by e-tplus** [comment]
Yes, how? Seems Debian 9 is no longer supported by Google Cloud:

`Ign:2 http://security.debian.org stretch/updates InRelease
Ign:3 http://deb.debian.org/debian stretch-updates InRelease
Err:4 http://security.debian.org stretch/updates Release
  404  Not Found
Get:5 http://packages.cloud.google.com/apt cloud-sdk-stretch InRelease [6,401 B]
Ign:6 http://deb.debian.org/debian stretch-backports InRelease
Err:7 http://deb.debian.org/debian stretch Release                       
  404  Not Found
Err:8 http://deb.debian.org/debian stretch-updates Release               
  404  Not Found
Get:9 http://packages.cloud.google.com/apt google-compute-engine-stretch-stable InRelease [5,141 B]
Err:10 http://deb.debian.org/debian stretch-backports Release                        
  404  Not Found
Get:11 http://packages.cloud.google.com/apt google-cloud-packages-archive-keyring-stretch InRelease [5,165 B]
Err:5 http://packages.cloud.google.com/apt cloud-sdk-stretch InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
Err:9 http://packages.cloud.google.com/apt google-compute-engine-stretch-stable InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
Err:11 http://packages.cloud.google.com/apt google-cloud-packages-archive-keyring-stretch InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
Get:12 https://download.docker.com/linux/debian stretch InRelease [44.8 kB]
Get:13 https://packages.cloud.google.com/apt google-cloud-ops-agent-stretch-all InRelease [5,092 B]
Err:13 https://packages.cloud.google.com/apt google-cloud-ops-agent-stretch-all InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
Reading package lists... Done 
E: The repository 'http://security.debian.org stretch/updates Release' does no longer have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://deb.debian.org/debian stretch Release' does no longer have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://deb.debian.org/debian stretch-updates Release' does no longer have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://deb.debian.org/debian stretch-backports Release' does no longer have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://packages.cloud.google.com/apt cloud-sdk-stretch InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://packages.cloud.google.com/apt google-compute-engine-stretch-stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://packages.cloud.google.com/apt google-cloud-packages-archive-keyring-stretch InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
W: GPG error: https://packages.cloud.google.com/apt google-cloud-ops-agent-stretch-all InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
E: The repository 'https://packages.cloud.google.com/apt google-cloud-ops-agent-stretch-all InRelease' is no longer signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.`

**Reply by cwanja** [reply]
You would need to address how to upgrade the underlying OS and then upgrade Docker. We do not support troubleshooting gCloud VMs. My guess would be you upgrade Debian. If you need assistance, you would need to reach out to Google or a local forum for them.
